sdk/src/main/java/com/mapbox/maps/Style.kt

```diff
@@ -20,24 +21,22 @@ import java.nio.ByteBuffer
  * @property pixelRatio the scale ratio of the style, default the device pixel ratio
  */
 class Style internal constructor(
-  private var styleManager: StyleManagerInterface?,
+  private var styleManager: StyleManagerInterface,
   override val pixelRatio: Float
 ) : StyleInterface {
 
-  internal fun onDestroy() {
-    styleManager = null
-  }
+  private var isMapValid = true
 
-  private fun getStyleManager(): StyleManagerInterface {
-    return styleManager ?: throw MapboxMapMemoryLeakException()
+  internal fun onMapViewDestroyed() {
+    isMapValid = false
   }
 
   /**
-   * Whether the Style instance is valid. Style will be invalid after MapView is destroyed or a new style is loaded and
-   * accessing Style will throw [MapboxMapMemoryLeakException].
+   * Whether the Style instance is valid.
+   * Style will be invalid after MapView is destroyed or a new style is loaded.
```

Comment:
Is it correct about the `or a new style is loaded.` ?

Reply:
it's interesting but as per current impl it's indeed not true, I'll rework it

Reply:
created follow-up task and fixed comment here to reflect true state of things